backend/fms_core/template_importer/row_handlers/sample_qc_spark/sample.py

```diff
@@ -0,0 +1,113 @@
+            barcode=sample['container']['barcode'],
+            coordinates=sample['coordinates'],
+        )
+
+        if sample_obj:
+            # Check if sample is not a library or a pool of libraries
+            if sample_obj.is_library:
+                self.errors['source_sample'] = f"Source sample can't be a library or a pool of libraries."
+
+            # Update sample with sample_information
+            if sample_information['concentration'] is None:
+                self.errors['concentration'] = 'Concentration value is required'
+
+            _, self.errors['sample_update'], self.warnings['sample_update'] = \
+                update_sample(sample_to_update=sample_obj,
+                              concentration=sample_information['concentration'])
+
+            # Update the sample's flags with sample information
+            _, self.errors['flags'], self.warnings['flags'] = update_qc_flags(sample=sample_obj,
+                                                                              quantity_flag=sample_information['quantity_flag'],
+                                                                              quality_flag=sample_information['quality_flag'])
+
+            process_measurement_obj, self.errors['process_measurement'], self.warnings['process_measurement'] = \
+                create_process_measurement(
+                    process=process_measurement['process'],
+                    source_sample=sample_obj,
+                    execution_date=process_measurement['execution_date'],
+                    volume_used=process_measurement['volume_used'],
+                    comment=process_measurement['comment'],
+                )
+
+            # Create process measurement's properties
+            if process_measurement_obj:
+                properties_obj, self.errors['properties'], self.warnings['properties'] = create_process_measurement_properties(
+                    process_measurement_properties,
+                    process_measurement_obj)
+
+                # Process the workflow action
+                self.errors['workflow'], self.warnings['workflow'] = execute_workflow_action(workflow_action=workflow["step_action"],
+                                                                                             step=workflow["step"],
+                                                                                             current_sample=sample_obj,
+                                                                                             process_measurement=process_measurement_obj)
+
+            if process_measurement_obj and properties_obj:
+                # Validate instruments according to platform
+                for instrument in INSTRUMENT_PROPERTIES:
+                    try:
+                        type = process_measurement_properties[instrument]['value']
+                        it = InstrumentType.objects.get(type=type)
```

Comment:
try using readable variable names as much as possible (instrument_type instead of it).

Reply:
i just copy pasted what was in sample qc row handler ;)